wincert.go

```diff
@@ -252,14 +252,25 @@ func isSelfSignedCert(cert *x509.Certificate) bool {
 	return certLikelyIssued(cert, cert)
 }
 
+func certsMatch(x, y *x509.Certificate) bool {
+	return bytes.Equal(x.RawSubject, y.RawSubject) &&
+		bytes.Equal(x.SubjectKeyId, y.SubjectKeyId) &&
+		x.SignatureAlgorithm == y.SignatureAlgorithm &&
+		bytes.Equal(x.RawIssuer, y.RawIssuer) &&
+		bytes.Equal(x.AuthorityKeyId, y.AuthorityKeyId) &&
+		x.PublicKeyAlgorithm == y.PublicKeyAlgorithm
+}
+
 func buildCertChains(trusted *x509.Certificate, untrusted []*x509.Certificate, chain []*x509.Certificate, depth *int) (chains [][]*x509.Certificate) {
-	alreadyInChain := func(x *x509.Certificate) bool {
-		for _, c := range chain {
-			if c.Equal(x) {
-				return true
+	removeCert := func(certs []*x509.Certificate, x *x509.Certificate) []*x509.Certificate {
+		newCerts := make([]*x509.Certificate, 0)
```

Comment:
Just a style thing.

```
	removeCert := func(certs []*x509.Certificate, x *x509.Certificate) (newCerts []*x509.Certificate) {
```

Or `var newCerts []*x509.Certificate`.

I have been told to not call `make` for empty slices since nil slices behave the same.